### PR TITLE
Build with ko

### PR DESCRIPTION
**Pull request by adambkaplan**
# Changes

- Add ko as a tool to build the container image
- Add contributing and developer docs
- Add an e2e placeholder for CI testing

/kind documentation

# Submitter Checklist

- [x] Includes tests if functionality changed/was added
- [x] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [x] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/master/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -106,11 +107,20 @@ verify-generate: generate
...
+	$(CONTAINER_ENGINE) build -t "${IMAGE_REPO}/operator:${TAG}" .
 
 # Push the docker image
 image-push:
-	$(CONTAINER_ENGINE) push ${IMG}
+	$(CONTAINER_ENGINE) push "${IMAGE_REPO}/operator:${TAG}"
+
+KO_DEST = $(shell pwd)/bin
+KO = $(KO_DEST)/ko
+ko:
```
Comment (ImJasonH) [inline Makefile:118]: Maybe rename this to `install-ko` ?

File: Dockerfile [hunk 2]
```diff
@@ -10,12 +10,12 @@ COPY go.sum go.sum
 RUN go mod download
```
Comment (ImJasonH) [inline Dockerfile:10]: Is this Dockerfile necessary at all anymore?

File: Makefile [hunk 3]
```diff
@@ -50,11 +51,11 @@ test: generate fmt vet manifests
 
 # Build manager binary
 manager: generate fmt vet
```
Comment (ImJasonH) [inline Makefile:53]: Should this be `operator` instead? It's a bit weird to build `./cmd/operator` into a binary with another name.

File: Makefile [hunk 4]
```diff
@@ -106,11 +107,20 @@ verify-generate: generate
 
 # Build the container image
 image-build: test
```
Comment (ImJasonH) [inline Makefile:109]: Do we need these at all anymore?

**Review by ImJasonH — COMMENTED** [review]
Nice! Just a few small things for consistency, but otherwise lgtm

File: Makefile (hunk 3 above)
Comment (adambkaplan) [reply Makefile:53]: agree, will need to shift a few other things around for consistency

File: Makefile (hunk 4 above)
Comment (adambkaplan) [reply Makefile:109]: With ko, don't think so. Same goes for the Dockerfile.

File: Makefile (hunk 1 above)
Comment (adambkaplan) [reply Makefile:112]: This is consistent with the other "install" targets (`operator-sdk`, `controller-gen`, `kustomize`).

**Review by adambkaplan — COMMENTED** [review]
/approve

Self-approving

**Review by ImJasonH — APPROVED** [review]

**Pull request merged**